frame/contracts/src/exec.rs

```diff
@@ -831,14 +834,19 @@ where
 		let do_transaction = || {
 			// We need to charge the storage deposit before the initial transfer so that
 			// it can create the account in case the initial transfer is < ed.
-			if entry_point == ExportedFunction::Constructor {
-				let frame = top_frame_mut!(self);
-				frame.nested_storage.charge_instantiate(
-					&self.origin,
-					&frame.account_id,
-					frame.contract_info.get(&frame.account_id),
-				)?;
-			}
+			// Also, we can only charge the storage deposit if there is an account id associated
```

Comment:
This sentence is hard to read/understand

Reply:
This bit has been refactored